### PR TITLE
Removing JCR leakage

### PR DESCRIPTION
**Pull request by escowles**
* Wrapping RepositoryException in RepositoryRuntimeException
* Using FedoraResource.replaceProperties instead of direct node updates
Addresses https://jira.duraspace.org/browse/FCREPO-1489

### CODE REVIEW

**Comment by awoods** [comment]
I assume this PR is dependent on: https://github.com/fcrepo4/fcrepo4/pull/781

**Comment by escowles** [comment]
Yes, this PR depends on fcrep4/fcrepo4#781

File: src/main/java/org/fcrepo/audit/InternalAuditor.java [hunk 1]
```diff
@@ -177,42 +208,76 @@ public void releaseConnections() {
...
+            m.add(createStatement(s, RDF_TYPE, createResource(PROV_EVENT)));
+            m.add(createStatement(s, PREMIS_TIME, createTypedLiteral(eventDate, XSDdateTime)));
+            m.add(createStatement(s, PREMIS_AGENT, createTypedLiteral(userID, XSDstring)));
+            m.add(createStatement(s, PREMIS_AGENT, createTypedLiteral(userAgent, XSDstring)));
+            if (auditEventType != null) {
+                m.add(createStatement(s, PREMIS_TYPE, createResource(auditEventType)));
+            }
+
+            final HttpResourceConverter subjects = new HttpResourceConverter(session, UriBuilder.fromUri(baseURL));
+            auditResource.replaceProperties(subjects, m, new RdfStream(), containerService);
```
Comment (escowles) [inline src/main/java/org/fcrepo/audit/InternalAuditor.java]: This isn't working for some reason -- I'm digging around in the RDF machinery trying to figure out why, but no luck so far.  I did fix the HttpResourceConvertor to use the audit resource URI for UriBuilder.forUri(), but that didn't fix it.

**Comment by escowles** [comment]
I put this in the JIRA ticket, but thought it would be good to have here to: This PR depends on https://github.com/fcrepo4/fcrepo4/pull/781 (adding FedoraResource.setURIProperty method) and https://github.com/fcrepo4/fcrepo4/pull/782 (updating DefaultIdentifierTranslator)

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4-labs/fcrepo-audit/commit/ace483a79fdc7c119ff90dfec6e8eab1814db07b

**Pull request closed without merging**